src/backend/src/services/change-requests.services.ts

```diff
@@ -621,4 +622,57 @@ export default class ChangeRequestsService {
       data: { dateDeleted: new Date(), deletedBy: { connect: { userId: submitter.userId } } }
     });
   }
+
+  /**
+   * set reviewers to the given change request
+   * @param submitter The user who sets a reviewer to the change request
+   * @param userIds The requested reviewers on the change request
+   * @param crId The change request that will be reviewed
+   */
+  static async requestCRReview(submitter: User, userIds: number[], crId: number) {
+    const reviewers = await getUsersSettings(userIds);
+
+    // check if any reviewers' role is below leadership
+    const underLeads = reviewers.filter((user) => !isLeadership(user.role));
+    const underLeadsNames = underLeads.map((reviewer) => reviewer.firstName + ' ' + reviewer.lastName);
+
+    if (underLeads.length > 0)
+      throw new AccessDeniedException(
+        `User(s) with the following names are not at least in a leadership: ${underLeadsNames.join(', ')}`
+      );
+
+    const foundCR = await prisma.change_Request.findUnique({
+      where: { crId },
+      ...changeRequestQueryArgs
+    });
+
+    if (!foundCR) throw new NotFoundException('Change Request', crId);
+
+    if (foundCR.submitterId !== submitter.userId)
+      throw new AccessDeniedException(`Only the author of this change request can request a reviewer`);
+
+    if (foundCR.dateDeleted) throw new DeletedException('Change Request', crId);
+
+    if (foundCR.reviewerId) throw new HttpException(400, `Cannot request a review on an already reviewed change request`);
+
+    const reviewerIds = reviewers.map((reviewer) => {
```

Comment:
do you need tit to be an object like this, or can you just map them to be (rev) => reviewer.userId

Reply:
Yeap, set in Prisma update requires to give a type of UserWhereUniqueInput[] (object of userId) instead of array of userId